### PR TITLE
Add subsource service

### PR DESCRIPTION
**Pull request by xZetsubou**
Hello, thanks for this addon it was really helpful one however ever since subscene shutdown the search no longer pull the subtitles for me especially "for anime" SubDL is quite good but not always e.g. old episodes sometimes isn't exists. 

These changes add SubSource service into a4ksubtitles.

### CODE REVIEW

File: .github/workflows/cron-tests-subsource.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: SubSource
+on:
+  push:
+    branches:
+      - add_subsource_tests
```
Comment (newt-sc) [inline .github/workflows/cron-tests-subsource.yml:5]: should be master

**Comment by newt-sc** [comment]
@xZetsubou Thank you for the contribution! 
Since yours is kind of the first with bigger changes, it will be good to pass the CI checks before merging. 

Every merge in master releases a new version of the addon. The commit in the PR should be only one and named with `release: vx.x.x`. In your case it should be `release: v3.16.0` This is used by a CI job to updated the releases.

You should update the version in addon.xml and add a change log in the same file.
Make sure to configure the git hooks in order for new hash to be generated. This is automated with a pre-commit hook:
```
git config core.hooksPath .githooks
```

When you configure the hook, you can `git commit --amend` without changes to trigger the hook.

File: .github/workflows/cron-tests-subsource.yml (hunk 1 above)
Comment (xZetsubou) [reply .github/workflows/cron-tests-subsource.yml:5]: Oops forgot about this thanks.

**Comment by xZetsubou** [comment]
I think I messed the commits a little bit, wouldn't squashing the PR handle this? 

I should've made this commits locally instead of GH workspace 😓

**Comment by newt-sc** [comment]
Yes it should be a single commit `release: v3.16.0` and it will work. You could just do:
```
git reset --soft master && git add -A && git commit -m 'release: v3.16.0' && git push --force
```

Also check the hook script, it must have given you some error (probably missing some dependency) as the `packages/addons.xml.crc` is currently being deleted.

**Comment by newt-sc** [comment]
OpenSubtitles tests were failing because they require username and password and those secrets are not available in the workflow run from fork-based PR. Some settings may need changing to make that work. But all should be fine.

Your changes are now available in the latest version of the addon.

**Comment by newt-sc** [comment]
The generated crc was wrong. Test the commands on master branch without any changes and verify the generated crc is the same:
```
cat packages/addons.xml | shasum | awk '{print $1}' > packages/addons.xml.crc && truncate -s -1 packages/addons.xml.crc && git status | grep 'nothing to commit'
```

I've released 3.16.1 with the fixed crc, otherwise KODI would not download the new version.

**Comment by SoulReaverFan** [comment]
> Hello, thanks for this addon it was really helpful one however ever since subscene shutdown the search no longer pull the subtitles for me especially "for anime" SubDL is quite good but not always e.g. old episodes sometimes isn't exists.
> 
> These changes add SubSource service into a4ksubtitles.

Awesome! Will it be possible to add Subscene.best support as well? That would make this add-on the best and only subtitles add-on needed I believe. Thanks so much for your contribution 👍

**Comment by xZetsubou** [comment]
> The generated crc was wrong. Test the commands on master branch without any changes and verify the generated crc is the same:

Did I do something wrong that makes generates wrong crc? How does Kodi detect new versions does it rely on "GH" releases?


@SoulReaverFan I don't think there's any benefits of adding "subscene" alternate domains I think SubSource and SubDL are now the way to go.

**Comment by newt-sc** [comment]
@xZetsubou
> Did I do something wrong that makes generates wrong crc?

The repo points to the packages/ addon.xml and addon.xml.crc: https://github.com/a4k-openproject/a4kSubtitles/blob/repository/addon.xml#L8

KODI checks them in order to know whether to fetch a new version. From the [docs](https://kodi.wiki/view/Add-on_repositories#checksum) it shouldn't matter what the crc is. But after merging your PR it did not detect it (may have been cache), that triggered me to check the crc and it generated a different one.

> How does Kodi detect new versions does it rely on "GH" releases?

The releases are pointing to tags with specific names which allows us to use the GitHub's archive API. The addon repo's datadir property is pointing to `https://github.com/a4k-openproject/a4kSubtitles/archive`. KODI gets that and suffixes it with `/addon.id/addon.id-x.y.z.zip.`. Which makes a valid url to create an archive from the tag in GitHub.

This way we bypass the whole create and upload zip files in GitHub repos as most other KODI repositories do.

**Comment by SoulReaverFan** [comment]
> > The generated crc was wrong. Test the commands on master branch without any changes and verify the generated crc is the same:
> 
> Did I do something wrong that makes generates wrong crc? How does Kodi detect new versions does it rely on "GH" releases?
> 
> @SoulReaverFan I don't think there's any benefits of adding "subscene" alternate domains I think SubSource and SubDL are now the way to go.

I think it works beautifully. It’s my go-to site and I believe it’s generally accepted as just being the new Subscene, so maybe could just be explained as re-adding Subscene support with updated url to make it work again like in the good old days! 🙂 I would really appreciate it and I know many other people online who would as well, but I also thank you very much for what you already have contributed to the project 👍 have a great day 🙂

**Pull request merged**